### DISCUSSION TITLE
Required One-to-One relation with cascade ondelete?

### DISCUSSION THREAD

**Discussion by ghost** [prisma/prisma#12866] [Q&A] [answered]
Hello, in this case how can I set:

the color requires name translation while also apply cascade on delete (delete a color also delete its name translation)

according to the docs, **_the side of the relation without a relation scalar (the field representing the foreign key in the database) must be optional_**.

```graphql
model ProductVariantColor {
  /// @HideField({match: ['ProductVariantColorCreate*Input', 'ProductVariantColorUpdate*Input']})
  id              String                          @id @default(uuid())
  languageCode    LanguageCode
  name            ProductVariantColorTranslation?
  value           String
  productVariants ProductVariant[]

  @@unique([languageCode, value])
}

model ProductVariantColorTranslation {
  /// @HideField({match: ['ProductVariantColorTranslationCreate*Input', 'ProductVariantColorTranslationUpdate*Input']})
  id                    String              @id @default(uuid())
  languageCode          LanguageCode
  value                 String
  productVariantColor   ProductVariantColor @relation(fields: [productVariantColorId], references: [id], onDelete: Cascade)
  productVariantColorId String              @unique
}
```

**Comment by austincrim** [comment, marked as answer]
Hey @lodisy 👋 ,

It's not possible to do this because we can't enforce this constraint at the database level. You will need to make sure the field is required in your app code. You could write a [Prisma middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware) to centralize that validation logic.

**Reply by ghost** [reply]
Thanks for your explanation
